### PR TITLE
A user-friendly guide for making the demonstrations metadata files

### PR DESCRIPTION
**Pull request by BM7878**
We already have a more technical description of the metadata files - this markdown file is just an explanation of what they are for people who are writing a demo and need to create a metadata file for it.

### CODE REVIEW

File: demonstrations_metadata_guide.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Demonstrations Metadata Guide
+
+This document describes how to create the metadata file when writing a new demonstration.
+
+## What is the metadata file any why do we have it?
+
+Each demonstration has an associated metadata file. This file is a JSON file; it has the same name, but a different ending - always `.metadata.json`.
```
Comment (josh146) [inline demonstrations_metadata_guide.md:7]: Might be good to mention the file location as well

File: demonstrations_metadata_guide.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+This file contains information that needs to be picked up by other systems and services. This is information like the title of the demonstration, the categories it's in, and the description and thumbnail image. By placing this information in a JSON file, it's easier for other systems - such as (and primarily) the pennylane.ai React website - to know what demonstrations exist, where they should be placed on the site, and how they are connected. You can think of the metadata file as 'registering' the demonstration with the system.
```
Comment (josh146) [inline demonstrations_metadata_guide.md:9]: Nice, definitely a good idea to motivate _why_ this is needed!

File: demonstrations_metadata_guide.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+* The `title` field should exactly match the title given in the `.py` file.
+* The `authors` field is a list of JSON objects, each of which has an `id`. This `id` field must exactly match the file name for the author in `_static/authors/`.
+* The `dateOfPublication` and `dateOfLastModification` fields must be in the form `YYYY-MM-DDTHH:MM:SS`, but the hour, minute, and second values can just be set to `0`.
+* The `categories` field is a list of strings. These strings will determine which categories the demonstration shows up in on the live site. The category string must exactly match that given for the title in `demonstrations/demonstrations_categories.metadata.json`.
+* You can leave the `tags` field empty for now.
+* The `previewImages` field is a list of images used in search results and other listings for the demonstration. One must be the default thumbnail to use, and must have the type `thumbnail`. Another can have the type `hero_image`, which is generally a larger version of the image. In all cases, the `uri` field must be a relative path to the image from the root of this repository.
```
Comment (josh146) [inline demonstrations_metadata_guide.md:22]: Do we need to document where these image files should live in terms of the repo structure?

**Review by josh146 — CHANGES_REQUESTED** [review]
Thanks Ben! Just a couple of minor comments

File: demonstrations_metadata_guide.md (hunk 2 above)
Comment (ikurecic) [inline demonstrations_metadata_guide.md:9]:
```suggestion
This file contains information that needs to be picked up by other systems and services. This is information like the title of the demonstration, the categories it's in, and the description and thumbnail image. By placing this information in a JSON file, it's easier for other systems — such as (and primarily) the pennylane.ai React website — to know what demonstrations exist, where they should be placed on the site, and how they are connected. You can think of the metadata file as 'registering' the demonstration with the system.
```
😉

File: demonstrations_metadata_guide.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+## How do you create the metadata file?
+
+Generally, the simplest thing to do is to copy one of the existing metadata files and change the data in it. Some files may not use all of the fields that they can, so for a complete list of available fields, see [this page](https://github.com/PennyLaneAI/qml/blob/master/demonstrations_metadata.md).
+
+Most of the fields should be self-explanatory, however, below is a list of things to remember while filling them in.
+
+* The `title` field should exactly match the title given in the `.py` file.
+* The `authors` field is a list of JSON objects, each of which has an `id`. This `id` field must exactly match the file name for the author in `_static/authors/`.
+* The `dateOfPublication` and `dateOfLastModification` fields must be in the form `YYYY-MM-DDTHH:MM:SS`, but the hour, minute, and second values can just be set to `0`.
+* The `categories` field is a list of strings. These strings will determine which categories the demonstration shows up in on the live site. The category string must exactly match that given for the title in `demonstrations/demonstrations_categories.metadata.json`.
```
Comment (ikurecic) [inline demonstrations_metadata_guide.md:20]: Are we limiting one demo to one category? If so, might also mention it here.

File: demonstrations_metadata_guide.md (hunk 2 above)
Comment (ikurecic) [reply demonstrations_metadata_guide.md:9]: I would also think about adding a piece of tantalizing info there, stress that adding these data will allow more people to find their demo in the vast depths of the internet (or the pennylane.ai website).

File: demonstrations_metadata_guide.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+* You can leave the `tags` field empty for now.
+* The `previewImages` field is a list of images used in search results and other listings for the demonstration. One must be the default thumbnail to use, and must have the type `thumbnail`. Another can have the type `hero_image`, which is generally a larger version of the image. In all cases, the `uri` field must be a relative path to the image from the root of this repository.
+* The `seoDescription` field should be the same as the description given in the Python file. This should ideally be close to 150 characters long (but not over), and end with a full stop.
+* The `doi` field can be left empty for now.
+* The `canonicalURL` field must be `/qml/demos/` + the file name of the demo without the `.py` ending.
```
Comment (ikurecic) [inline demonstrations_metadata_guide.md:25]: By this point in the file I'm starting to feel like there's a lof of opportunity for a lapsus. Would it be better to have a fixed example to refer to?

**Review by ikurecic — COMMENTED** [review]
Looks good (added just one suggestion, thanks ☺), but I'm struggling to follow the naming conventions at some points. Could you maybe add an example and follow it? Even just inserting a name instead of naming extensions would make it easier to orient yourself.

File: demonstrations_metadata_guide.md (hunk 5 above)
Comment (josh146) [reply demonstrations_metadata_guide.md:25]: Never heard the word lapsus before!

File: demonstrations_metadata_guide.md (hunk 1 above)
Comment (BM7878) [reply demonstrations_metadata_guide.md]: Added.

File: demonstrations_metadata_guide.md (hunk 4 above)
Comment (BM7878) [reply demonstrations_metadata_guide.md:20]: Multiple categories are now allowed.

File: demonstrations_metadata_guide.md (hunk 2 above)
Comment (BM7878) [reply demonstrations_metadata_guide.md]: Good idea - will do.

File: demonstrations_metadata_guide.md (hunk 3 above)
Comment (BM7878) [reply demonstrations_metadata_guide.md:22]: Added

**Pull request closed without merging**